src/Terrain/terrain_mesh.hpp

```diff
@@ -0,0 +1,104 @@
+//! This is a test. The content tent of this class should be transferred into chunk
+
+#include <vector>
+
+// Include GLEW
+#include <GL/glew.h>
+
+// Include GLFW
+#include <GLFW/glfw3.h>
+
+// Include GLM
+#include <glm/glm.hpp>
+
+#include "../GUI/meshloader.hpp"
+
+#ifndef __TERRAIN_MESH_HPP__
+#define __TERRAIN_MESH_HPP__
+
+class TerrainMesh : public MeshLoader::SingleComplexMesh {
+private:
+    GLuint vertex_buffer_;
+    GLuint color_buffer_;
+    GLuint normal_buffer_;
+    GLuint element_buffer_;
+    unsigned int num_vertices_;
+public:
+    TerrainMesh(const TerrainMesh& obj){
+        vertex_buffer_ = obj.vertex_buffer_;
+        color_buffer_ = obj.color_buffer_;
+        normal_buffer_ = obj.normal_buffer_;
+        element_buffer_ = obj.element_buffer_;
+        num_vertices_ = obj.num_vertices_;
+    };
+    TerrainMesh& operator=(const TerrainMesh& obj){
+        vertex_buffer_ = obj.vertex_buffer_;
+        color_buffer_ = obj.color_buffer_;
+        normal_buffer_ = obj.normal_buffer_;
+        element_buffer_ = obj.element_buffer_;
+        num_vertices_ = obj.num_vertices_;
+        return *this;
+    }
+
+    TerrainMesh(){};
+
+    TerrainMesh(std::vector<unsigned short> &indices,
```

Comment:
How do you make sure this doesn't run multiple times.

Reply:
that's not bad